### PR TITLE
KIALI-1671 Refactor validation endpoints

### PR DESCRIPTION
**Pull request by burmanm**
** Describe the change **

This is counterpart for the kiali PR 749. Both must be merged at the same time as otherwise neither will function correctly. This removes the usage validation REST-endpoints and instead fetches validation results as part of the same details request. No UI-changes should happen.

I probably broke something while refactoring, so comments are welcome. Also, any additional stuff I could remove from the codebase would be nice to know.

** Issue reference **

KIALI-1671

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "@kiali/kiali-ui",
   "version": "0.12.0",
+  "proxy": "https://kiali-istio-system.192.168.1.105.nip.io",
   "description": "React UI for [Kiali](https://github.com/kiali/kiali).",
```
Comment (aljesusg) [inline package.json]: You must remove proxy line

File: src/components/ConfigValidation/ConfigIndicator.tsx [hunk 2]
```diff
@@ -57,7 +57,9 @@ export class ConfigIndicator extends React.PureComponent<Props, {}> {
   numberOfChecks = (type: string) => {
     let numCheck = 0;
     this.props.validations.forEach(validation => {
-      numCheck += validation.checks.filter(i => i.severity === type).length;
+      if (validation !== undefined) {
```
Comment (lucasponce) [inline src/components/ConfigValidation/ConfigIndicator.tsx]: I wonder whether
```
if (validation) { ... }
```
But I wonder if the mapping of `this.props.validations` cannot force to have non undefined values.

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx [hunk 3]
```diff
@@ -56,21 +54,13 @@ class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioCo
           props.objectSubtype,
           props.object
         )
-      : API.getIstioConfigDetail(authentication(), props.namespace, props.objectType, props.object);
+      : API.getIstioConfigDetail(authentication(), props.namespace, props.objectType, props.object, true);
 
     // Note that adapters/templates are not supported yet for validations
-    // This logic will be refactored later on KIALI-1671
```
Comment (lucasponce) [inline src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx]: Cool :-)

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx [hunk 4]
```diff
@@ -121,7 +111,10 @@ class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioCo
 
   renderEditor = (istioObject: any) => {
     const yamlSource = yaml.safeDump(istioObject, safeDumpOptions);
-    const aceValidations = parseAceValidations(yamlSource, this.state.validations);
+    const aceValidations = this.state.istioObjectDetails
+      ? parseAceValidations(yamlSource, this.state.istioObjectDetails.validation)
+      : parseAceValidations(yamlSource, undefined);
```
Comment (lucasponce) [inline src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx]: This looks odd, if not defined this.state.istioObjectDetails perhaps you can build an empty aceValidations object instead to invoke the parseAceValidations() with undefined.

File: src/services/__mockData__/getIstioConfigValidations.json [hunk 5]
```diff
@@ -1,3 +1,4 @@
+// TODO Combine with getIstioConfigDetail
```
Comment (lucasponce) [inline src/services/__mockData__/getIstioConfigValidations.json]: Missing test ?

**Comment by lucasponce** [comment]
Main code looks ok.
Some comments but no blocker.
Which needs to be investigated is the issue with metrics, just to see if this PR needs some rebase or something is out of the radar with the backend.

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx (hunk 4 above)
Comment (burmanm) [reply src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx]: Well, the ``parseAceValidations()`` appeared to expect such behavior as it has ``validation?: ObjectValidation`` in the parameter list as well as ``if (!validation`` branch. So given that behavior I think the choice makes sense.

File: src/components/ConfigValidation/ConfigIndicator.tsx (hunk 2 above)
Comment (burmanm) [reply src/components/ConfigValidation/ConfigIndicator.tsx]: With backend change always returning empty one, this check isn't needed anymore.

**Comment by lucasponce** [comment]
I wanted to merge this one but github didn't put an invalid mark on this PR on the conflict.
Please @burmanm could you rebase ASAP ? as the backend PR would make fails the current master.
Thanks !

**Pull request merged**